Keep focus for Windows and Linux

ig0774 opened the pull request:
This is an adaptation and generalisation of @r-stein's suggestion from #495. Basically if `keep_focus` is set on Windows or Linux, it runs either `sublime_text` or `subl` to regain focus after a (configurable) short delay.

I've added some code to work out where the appropriate executable is, though it should be auto-detected on *most* configurations. The only worrying setup is ST2 where `subl` is not available on the `PATH`, in which, case the platform-specific `sublime_executable` setting must be set appropriately. Alternatively, the `sublime_executable` setting can be set and the auto-detect code will not be run.

r-stein commented on jumpToPDF.py:
```diff
@@ -12,10 +12,96 @@
+	if sublime.platform() == 'windows':
+		startupinfo = subprocess.STARTUPINFO()
+		startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+		shell = True
```
At least in my setup this must be
``` python
shell = _ST3
```
to work with ST2 (portable)

ig0774 replied:
I'm happy to change it, but it works on my ST2 on Windows 7 this way. My main reason for setting `shell = True` on Windows is that it runs the command through `cmd.exe` which tends to behave more like I expect. I didn't expect it would affect the outcome here either way.

r-stein commented:
I think it would be nice to add the default sublime installation path into `get_sublime_executable`, such that it works out of the box for the majority of users:
``` python
if sublime.platform() == "windows":
	if _ST3:
		DEFAULT_PATH = "C:\\Program Files\\Sublime Text 3\\sublime_text.exe"
	else:
		DEFAULT_PATH = "C:\\Program Files\\Sublime Text 2\\sublime_text.exe"
	if os.path.exists(DEFAULT_PATH):
		get_sublime_executable.result = DEFAULT_PATH
		return get_sublime_executable.result
```
*(Same for the default SumatraPDF installation path.)*

Limro commented:
Heads up, with those hard-coded paths. I know you said *majority*, but that is those with an english operating system.
Is it possible to ask the system for the default installation path?
I know PowerShell have two commands to find the installation paths:
```batch
$x64Path = "${env:ProgramFiles}"
$x86Path = "${env:ProgramFiles(x86)}"
```

r-stein commented:
Thanks @Limro. Although the OS language does not seem to effect the folder. Some user might change the default directory, e.g. on an other partition.

ig0774 commented:
Yeah... IIRC, on non-English installs there's a *hidden* `Program Files` folder and a localised hardlink to it that's localised to allow for software written by English-speakers who don't consider the possibity of a different name. Still it's (relatively) easy to makes use of the environment variables. I hadn't realised it wasn't automatically added to the `PATH`... Maybe I was thinking of ST3? In any case, I'll add in some checks.

BTW, I'm not as concerned with the default path for ST3 because the `sublime.executable_path()` function call should already tell us the executable of the version that is actually in use. It's only ST2 (where that function doesn't exist) where this becomes a problem.

ig0774 commented:
d143719 should address the default install path on Windows. I don't think its determinable for ST2 on Linux, but let me know if I'm wrong.

A couple of comments on what I've done: 
 1. It only checks for ST2 as per my comment above. 
 1. I only used `%PROGRAMFILES%` even though `%PROGRAMFILES(x86)%` might exist. The purpose of this code is to find the executable for the current running instance of ST2. For a 64-bit version on a 64-bit OS or a 32-bit version on a 32-bit OS (eep), this will always be the equivalent of `C:\Program Files` and for a 32-bit version on a 64-bit OS it will be the equivalent of `C:\Program Files (x86)`. In other words, it will get us the right folder for the current version of ST, regardless of its bittedness.

Thanks both @r-stein and @Limro for your helpful remarks! If I've missed anything else, let me know!

r-stein commented:
This could also be problematic if someone uses ST2, but has ST3 in the path. E.g. if I try it on a portable ST2 it jumps to my ST3 installation (which is in the path). This will not affect many users and can easily be fixed by changing the settings, but if you have an idea to fix this.

ig0774 commented:
Thanks. At least this should stop ST2 from running an ST3 executable.

r-stein commented:
Agree. If I change line `43` and line `124` to `shell = _ST3` it even works out of the box on my ST2 portable with windows.

r-stein commented:
Just a small remark: I think it would be more intuitive and user friendly to use seconds instead of ms as unit for the `"keep_focus_delay"` option and convert it to ms (`int(x * 1000)`).

ig0774 commented:
Ok. I think I've addressed all of @r-stein's very good points and I added some code with some default paths on Linux, but I think I've actually found a way to make this work without a lot of fuss on ST2, since it appears that `sys.executable` points to the full path of `sublime_text` or `sublime_text.exe`, which makes sense since ST2 appears to run a statically compiled version of Python.
